sklearn/manifold/spectral_embedding_.py

```diff
@@ -289,11 +289,25 @@ def spectral_embedding(adjacency, n_components=8, eigen_solver=None,
         laplacian = check_array(laplacian, dtype=np.float64,
                                 accept_sparse=True)
         laplacian = _set_diag(laplacian, 1, norm_laplacian)
+
+        # The Laplacian matrix is always singular, having at least one zero
+        # eigenvalue, corresponding to the trivial eigenvector, which is a
+        # constant. Using a singular matrix for preconditioning may result in
+        # random failures in LOBPCG and is not supported by the existing
+        # theory:
+        #     see https://doi.org/10.1007/s10208-015-9297-1
+        # Shift the Laplacian so its diagononal is not all ones. The shift
+        # does change the eigenpairs however, so we'll feed the shifted
+        # matrix to the solver and afterward set it back to the original.
+        laplacian = laplacian + 1e-5 * sparse.eye(laplacian.shape[0])
+        # noinspection PyUnboundLocalVariable
```

Comment:
I don't think this belongs here

Reply:
`# noinspection PyUnboundLocalVariable` 
surely needs to be removed